### ISSUE TITLE
egui layer shell example

### ISSUE THREAD

**Issue by yavko** [Smithay/client-toolkit#349]
Hello I'm new to sctk, and it would be helpful for me and others who want to use egui with layer shell if there was an example, thanks!

**Comment by m-hugo** [comment]
there's two way to go about it, both are a lot of work:

- go through winit, implement this https://github.com/rust-windowing/winit/issues/2582 and modify egui-winit to use the newly exposed api
-  modify egui-winit to use sctk directly instead of winit

**Comment by yavko** [comment]
> there's two way to go about it, both are a lot of work:
> 
>     * go through winit, implement this [Feature request: Support for Wayland's wlr-layer-shell protocol rust-windowing/winit#2582](https://github.com/rust-windowing/winit/issues/2582) and modify egui-winit to use the newly exposed api
> 
>     * modify egui-winit to use sctk directly instead of winit

Doesn't egui support using wgpu, or glow, could they be used with sctk?

**Comment by m-hugo** [comment]
yes, it shoud be easy to make egui render to a wayland window through wgpu, the 'hard' part is the windowing part

**Comment by yavko** [comment]
> yes, it shoud be easy to make egui render to a sctk wayland window through wgpu, the 'hard' part is the windowing 

So would it also be possible through layer shell?

**Comment by m-hugo** [comment]
yes your main would look like this https://github.com/Smithay/client-toolkit/blob/master/examples/simple_layer.rs / https://github.com/Smithay/client-toolkit/blob/master/examples/wgpu.rs
but instead of using wgpu direcly you'd use the egui stuff like https://github.com/rust-mobile/rust-android-examples/blob/main/agdk-egui/src/lib.rs

**Comment by ids1024** [comment]
The best solution for this would be layer shell support in winit (https://github.com/rust-windowing/winit/issues/2582). Or perhaps a winit-like library on top of sctk could provide a dedicated way to deal with layer shell surfaces.

**Comment by yavko** [comment]
> The best solution for this would be layer shell support in winit ([rust-windowing/winit#2582](https://github.com/rust-windowing/winit/issues/2582)). Or perhaps a winit-like library on top of sctk could provide a dedicated way to deal with layer shell surfaces.

I would prefer that too, but a example using raw apis would be cool for the moment, and could be useful.

**Comment by Ciantic** [comment]
Here is the **layer shell** example:

https://github.com/Ciantic/egui-wgpu-example-smithay/tree/layer-shell

Here is the **windowed** example:

https://github.com/Ciantic/egui-wgpu-example-smithay


<img width="787" height="566" alt="Image" src="https://github.com/user-attachments/assets/97a1f3e0-591d-4fb9-96bc-d51fe01ecdc1" />

What I'm annoyed is the memory usage, WGPU + Vulkan uses always at minimum ~100M of memory (USS)...

(Not fault of EGUI or smithay, WGPU's own examples also use that much.)
